plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionTransforms.java

```diff
@@ -263,24 +289,32 @@ private static Block extractTimestamp(Block block, LongUnaryOperator function)
         return builder.build();
     }
 
-    private static Block yearsFromTimestampWithTimeZone(Block block)
+    private static ColumnTransform yearsFromTimestampWithTimeZone()
     {
-        return extractTimestampWithTimeZone(block, PartitionTransforms::epochYear);
+        return new ColumnTransform(
+                INTEGER,
+                block -> extractTimestampWithTimeZone(block, PartitionTransforms::epochYear));
     }
 
-    private static Block monthsFromTimestampWithTimeZone(Block block)
+    private static ColumnTransform monthsFromTimestampWithTimeZone()
     {
-        return extractTimestampWithTimeZone(block, PartitionTransforms::epochMonth);
+        return new ColumnTransform(
+                INTEGER,
+                block -> extractTimestampWithTimeZone(block, PartitionTransforms::epochMonth));
     }
 
-    private static Block daysFromTimestampWithTimeZone(Block block)
+    private static ColumnTransform daysFromTimestampWithTimeZone()
     {
-        return extractTimestampWithTimeZone(block, PartitionTransforms::epochDay);
+        return new ColumnTransform(
+                INTEGER,
+                block -> extractTimestampWithTimeZone(block, PartitionTransforms::epochDay));
     }
 
-    private static Block hoursFromTimestampWithTimeZone(Block block)
+    private static ColumnTransform hoursFromTimestampWithTimeZone()
```

Comment:
Unrelated: 
You can make `ColumnTransform` package private

Reply:
>Unrelated:

 let me skip this for now